runtime/balrt.h

```diff
@@ -124,6 +129,19 @@ typedef struct {
     Tid tid;
 } StructureDesc, *StructureDescPtr;
 
+// This is the abstract version of filler descriptor. Each type must 
+// implement there own version (or reuse a common version such as GenericFillerDesc)
+typedef struct FillerDesc {
+    TaggedPtr (*create)(struct FillerDesc *fillerDesc, bool *hasIdentityPtr);
+} *FillerDescPtr;
+
+typedef struct GenericFillerDesc {
+    TaggedPtr (*create)(struct GenericFillerDesc *fillerDesc, bool *hasIdentityPtr);
+    TaggedPtr genericValue;
+} *GenericFillerDescPtr;
+
+extern TaggedPtr _bal_generic_filler_create(GenericFillerDescPtr fillerDesc, bool *hasIdentityPtr);
```

Comment:
This should be down below with all the other extern declarations.

Reply:
fixed